src/lint_config.rs

```diff
@@ -7,9 +7,12 @@ use log::debug;
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize)]
-pub struct LintRunnerConfig {
+pub struct LintRunnerConfig{
     #[serde(rename = "linter")]
     pub linters: Vec<LintConfig>,
+
+    #[serde(default)]
+    pub merge_base_with: String,
```

Comment:
This should probably be optional instead of using the empty string to represent nullity.

Reply:
I tried that first, setting this to type `Option<String>` but got stuck on some silly error in main.rs when I replaced line 216 with `RevisionOpt::MergeBaseWith(lint_runner_config.merge_base_with.expect("Should be defined")` I'd get the compiler error:

```
error[E0382]: borrow of partially moved value: `lint_runner_config`
   --> src/main.rs:254:56
    |
216 |         RevisionOpt::MergeBaseWith(lint_runner_config.merge_base_with.expect("Should be defined"))
    |                                    ---------------------------------- --------------------------- `lint_runner_config.merge_base_with` partially moved due to this method call
    |                                    |
    |                                    help: consider calling `.as_ref()` or `.as_mut()` to borrow the type's contents
...
254 |             check_init_changed(&persistent_data_store, &lint_runner_config)?;
    |                                                        ^^^^^^^^^^^^^^^^^^^ value borrowed here after partial move
    |
```

I've never used Rust before but this borrowing concept seems like it means I can't read the value of a string if there's a future line that reads that string by reference.

If you know a better, React-appropriate way to change that line, I'd be happy to do do that.

Reply:
Should be fixed by cloning `merge_base_with`, like:
```
lint_runner_config.merge_base_with.clone().expect("Should be defined"))
```